htdocs/compta/prelevement/class/bonprelevement.class.php

```diff
@@ -1552,7 +1605,7 @@ public function addNotification($db, $user, $action)
 	 */
 	public function generate($format = 'ALL', $executiondate = 0, $type = 'direct-debit', $fk_bank_account = 0, $user_dest = 0)
 	{
-		global $conf, $langs, $mysoc;
+		global $conf, $langs, $mysoc, $sourcetype;
```

Comment:
The global variable are reserved for variable that we need everywhere in code.Here the value $sourcetype should be propagated as a parameter in generate()